src/win/getnameinfo.c

```diff
@@ -38,11 +38,40 @@ int WSAAPI GetNameInfoW(
 );
 #endif
 
+static int uv__to_utf8_fixed_size(const WCHAR * src, char * dst, size_t dst_len) {
```

Comment:
I don't think you need to do this. We have `uv__convert_utf16_to_utf8()`, which mallocs the destination buffer for you. In this case, we already have the destination buffer and its size, so a single call to `WideCharToMultiByte()` should suffice (like in the existing code). `WideCharToMultiByte()` also returns an error if there isn't enough buffer space, so the first call and the length check shouldn't be needed.

Reply:
I didn't want to break the ABI/API. host and service are already allocated and not really private according this comment:
https://github.com/libuv/libuv/blob/619937c783a05b51fba95cc9a62543deeffe5fa7/include/uv.h#L845

Reply:
What I meant is that I don't think this helper function is needed. The existing code with the single call to `WideCharToMultiByte()` for each field should be enough. I'm not sure why there wasn't error checking in place in the existing code though.